docs/resource-providers/iam-role-providers.md

```diff
@@ -0,0 +1,27 @@
+## IAM Role Resource Providers
+
+### CreateRoleProvider
+Creates a new role based on the given role id, principal, and policies and provides it to the stack.
+
+Example Implementation:
+```typescript
+blueprints.EksBlueprint.builder()
+  .resourceProvider('service-role', 
+    new blueprints.CreateRoleProvider(
+      'ServiceRole', 
+      new iam.ServicePrincipal('ec2.amazonaws.com'), 
+      [iam.ManagedPolicy.fromAwsManagedPolicyName("AWSEC2SpotServiceRolePolicy")])).
+  ...
+  .build()
+```
+
+### LookupRoleProvider
+Looks up role by name and provides it to the stack.
```

Comment:
Let's call out that it will look it up in the account where blueprint is deployed. That means that if the same blueprint is deployed in 5 accounts, each should have that role defined.